### ISSUE TITLE
docs: Don't distinguish structs and tuple structs with only private fields

### ISSUE THREAD

**Issue by clarcharr** [rust-lang/rust#38101]
For example, [`AddrParseError`](https://doc.rust-lang.org/std/net/struct.AddrParseError.html) shows up as `pub struct AddrParseError(_);` and [`Ipv4Addr`](https://doc.rust-lang.org/std/net/struct.Ipv4Addr.html) shows up as `pub struct Ipv4Addr { /* fields omitted */ }`

I feel like all structs should show up as the latter form if all of the fields are private. It technically doesn't matter that the struct is a tuple struct if the fields are all private.

**Comment by bluss** [comment]
Yes, it is time, because the relevant part of #35626 is stable in nightly.

**Comment by steveklabnik** [comment]
Triage: this bug still reproduces

**Comment by clarfonthey** [comment]
This bug still reproduces, in case you were wondering.

**Comment by camelid** [comment]
Yeah, it's always bothered me that tuple structs with all private fields show how many fields they have, even though they're supposed to be fully private.

**Comment by clarfonthey** [comment]
The main benefit of the struct-based syntax is that it's actually valid Rust syntax, but if you don't care about that, the latter also works. I don't believe that there's any tuple-based syntax that works like that.

**Comment by camelid** [comment]
> The main benefit of the struct-based syntax is that it's actually valid Rust syntax, but if you don't care about that, the latter also works. I don't believe that there's any tuple-based syntax that works like that.

`Struct { 1: T }` doesn't seem to be valid syntax, unless you were referring to something else?

**Comment by clarfonthey** [comment]
So, I should have clarified: `Struct { 1: T }` isn't a valid struct definition, but `Struct { 1: x }` is a valid pattern. So, I guess what I meant to say was that it's *closer* to valid Rust syntax, although it's still invalid.
